### PR TITLE
evalresp: add routine to get response for specific frequencies

### PR DESCRIPTION
**Pull request by megies**
This adds a routine to request response for a set of discrete frequencies using evalresp. I'll also add a routine for magnitude estimation based on a response object now.

### CODE REVIEW

**Comment by megies** [comment]
With this PR the current `estimate_magnitude` function that is based on a poles-and-zeros dictionary can also work with response objects. I left the argument name `paz` as is and added to the existing API docstrings, as renaming the argument might break code or take a lot of time to get deprecation reroutes right..

The whole magnitude estimation could use some more/better tests, but at least there are some..

**Comment by megies** [comment]
This is ready for review/comments, all test fails are due to #1599.

File: obspy/signal/invsim.py [hunk 1]
```diff
@@ -273,26 +272,66 @@ def evalresp(t_samp, nfft, filename, date, station='*', channel='*',
         datime = C.create_string_buffer(
             date.format_seed().encode('ascii', 'strict'))
         fn = C.create_string_buffer(tempfile.encode('ascii', 'strict'))
-        nfreqs = C.c_int(freqs.shape[0])
+        frequencies = np.array(frequencies)
```
Comment (QuLogic) [inline obspy/signal/invsim.py]: `np.asarray` again?

File: obspy/signal/invsim.py [hunk 2]
```diff
@@ -637,16 +678,28 @@ def estimate_magnitude(paz, amplitude, timespan, h_dist):
...
+    wood_anderson_amplitude_functions = []
+    for i in paz:
+        if isinstance(i, dict):
+            wood_anderson_amplitude_functions.append(
+                estimate_wood_anderson_amplitude)
+        elif isinstance(i, Response):
+            wood_anderson_amplitude_functions.append(
+                estimate_wood_anderson_amplitude_using_response)
+        else:
+            raise TypeError()
```
Comment (QuLogic) [inline obspy/signal/invsim.py]: Add message.

File: obspy/core/inventory/response.py [hunk 3]
```diff
@@ -1001,25 +995,24 @@ def get_unit_mapping(key):
...
                 _amp_c = (C.c_double * rl.nresp)()
                 _phase_c = (C.c_double * rl.nresp)()
 
-                for i in range(len(freqs)):
-                    _freq_c[i] = freqs[i]
-                    _amp_c[i] = amp[i]
-                    _phase_c[i] = phase[i]
+                _freq_c[:] = frequencies[:]
+                _amp_c[:] = amp[:]
+                _phase_c[:] = phase[:]
```
Comment (krischer) [inline obspy/core/inventory/response.py]: Hmm. I cannot believe I did not try this the first time around. Lol - this should also be quite a bit faster now :-)

**Comment by krischer** [comment]
I added a changelog commit directly to the master: 7fd84f9ed2ab0ba8757d28ca5c8ba61dd98794d1

**Comment by megies** [comment]
Just noticed that..

```python
>>> d = AttribDict()
>>> isinstance(d, dict)
False
```

..which is quite unfortunate but I guess adding `dict` as a base class on `AttribDict` might be dangerous.. :-/

**Comment by claudiodsf** [comment]
You can always check that it is dictionary-like:
```python
>>> from collections import Mapping
>>> d = AttribDict()
>>> isinstance(d, Mapping)
True
```

**Pull request merged**